### PR TITLE
Update TypeScript and Eslint templates

### PR DESCRIPTION
**Pull request by devinea**
- Support latest eslint typescript versions https://github.com/SAP/open-ux-tools/issues/1719
- .babelrc needs to be removed
- all dependencies to babel etc needs to be removed in templates
- increase the devDependency version of UI5 CLI, typescript middleware to the latest
- update the setting of the transpile task/middleware in the ui5.yaml
- Fix issue in CAP projects eslint shows in text editor for fiori appp TS files. `Parsing error: Cannot read file '/users/xxxxxx/mycapapp/tsconfig.json'.eslint`
- https://github.com/SAP/open-ux-tools/issues/1751

### CODE REVIEW

File: packages/eslint-plugin-fiori-tools/eslintrc-typescript.js [hunk 1]
```diff
@@ -5,7 +5,17 @@ const overrides = [
     {
         'plugins': ['@typescript-eslint'],
         'files': [`./${sourceCodePath}/*.ts`, `./${sourceCodePath}/**/*.ts`],
-        'excludedFiles': ['*.d.ts', '**/*.d.ts'],
+        'excludedFiles': [
+            'target/**',
+            `${sourceCodePath}/test/changes_loader.ts`,
+            `${sourceCodePath}/test/changes_preview.ts`,
```
Comment (tobiasqueck) [inline packages/eslint-plugin-fiori-tools/eslintrc-typescript.js:11]: Why do we exclude these ones? Same for `${testCodePath}/**/Example.qunit.ts`

File: packages/eslint-plugin-fiori-tools/package.json [hunk 2]
```diff
@@ -25,8 +25,8 @@
     },
     "peerDependencies": {
         "eslint": ">=7",
-        "@typescript-eslint/eslint-plugin": "^5.59.0",
-        "@typescript-eslint/parser": "^5.59.0"
+        "@typescript-eslint/eslint-plugin": ">=5.59.0",
+        "@typescript-eslint/parser": ">=5.59.0"
```
Comment (tobiasqueck) [inline packages/eslint-plugin-fiori-tools/package.json:29]: I am always confused with `~` `^` and `>=`. Could you please explain why this change?

**Review by tobiasqueck — COMMENTED** [review]
* changeset good
* version requirement for `@ui5/cli` correctly increased
* `.babelrc` removed and required setting moved into the `ui5.yaml`
* didn't test locally but the snapshots look correct
* minor clarification questions open

File: packages/eslint-plugin-fiori-tools/package.json (hunk 2 above)
Comment (devinea) [reply packages/eslint-plugin-fiori-tools/package.json:29]: This is to address https://github.com/SAP/open-ux-tools/issues/1719#issuecomment-1976727749 
and set a minimum version of `"@typescript-eslint/*` so that it can be used with even newer major versions which are now available.

File: packages/eslint-plugin-fiori-tools/eslintrc-typescript.js (hunk 1 above)
Comment (devinea) [reply packages/eslint-plugin-fiori-tools/eslintrc-typescript.js:11]: These test files have eslint errors. But in some cases I think it we better clarity of example to keep these issues. `reject` for example. 

What do you think?

`changes_loader.ts`
```
open-ux-tools/packages/fiori-elements-writer/test/test-output/lropTemplates/lropV2_ts_ui5_1_108/webapp/test/changes_loader.ts
      16:50  error  'reject' is defined but never used  @typescript-eslint/no-unused-vars
      34:36  error  'obj' is defined but never used     @typescript-eslint/no-unused-vars
```

`changes_preview.ts`    
```
open-ux-tools/packages/fiori-elements-writer/test/test-output/lropTemplates/lropV2_ts_ui5_1_108/webapp/test/changes_preview.ts
        9:31  error  'oChange' is defined but never used      @typescript-eslint/no-unused-vars
       48:55  error  'reject' is defined but never used       @typescript-eslint/no-unused-vars
       82:76  error  'rejectInner' is defined but never used  @typescript-eslint/no-unused-vars
      109:61  error  'obj' is defined but never used          @typescript-eslint/no-unused-vars
```



`Example.qunit.ts` 
```   
open-ux-tools/packages/ui5-library-writer/test/test-output/com.sap.myui5tslib113/test/com/sap/myui5tslib113/qunit/Example.qunit.ts
      32:40  error  'reject' is defined but never used             @typescript-eslint/no-unused-vars
      54:9   error  'oExample' is assigned a value but never used  @typescript-eslint/no-unused-vars
      60:40  error  'reject' is defined but never used             @typescript-eslint/no-unused-vars
```

File: packages/eslint-plugin-fiori-tools/package.json (hunk 2 above)
Comment (tobiasqueck) [reply packages/eslint-plugin-fiori-tools/package.json:29]: Thanks.

File: packages/eslint-plugin-fiori-tools/package.json [hunk 3]
```diff
@@ -25,8 +25,8 @@
     },
```
Comment (tobiasqueck) [inline packages/eslint-plugin-fiori-tools/package.json:25]: @devinea 
The devDependency `eslint@7.32.0` has a depedency to `minimatch@3.0.4` which has a security issue. It looks like there is no newer `7.x` version with a fix. Could we switch to `8.x` e.g. `8.57.0`?

Comment (devinea) [reply packages/eslint-plugin-fiori-tools/package.json:25]: Yeah makes sense.

File: packages/eslint-plugin-fiori-tools/eslintrc-typescript.js (hunk 1 above)
Comment (tobiasqueck) [reply packages/eslint-plugin-fiori-tools/eslintrc-typescript.js:11]: I would have recommended to rename them to `_reject` etc. however, since we are going to get rid of these files anyhow (hopefully sooner than later) but still some projects have them, I understand why it is more convenient for consumers if we simply ignore them.

File: packages/eslint-plugin-fiori-tools/package.json (hunk 3 above)
Comment (tobiasqueck) [reply packages/eslint-plugin-fiori-tools/package.json:25]: Thanks

File: packages/eslint-plugin-fiori-tools/eslintrc-typescript.js [hunk 4]
```diff
@@ -5,11 +5,21 @@ const overrides = [
...
+            `${testCodePath}/**/Example.qunit.ts`,
+            'backup/**',
+            '*.d.ts',
+            '**/*.d.ts'
+        ],
         'parser': '@typescript-eslint/parser',
         'extends': ['plugin:@typescript-eslint/recommended'],
         'parserOptions': {
-            'project': ['./tsconfig.json']
+            'project': true // uses local tsconfig.json nearest to file being linted. Especially important for monorepos
```
Comment (donal-tobin-sap) [inline packages/eslint-plugin-fiori-tools/eslintrc-typescript.js:22]: Note.

**Review by donal-tobin-sap — APPROVED** [review]
Looks good to me for lock file changes.

**Comment by devinea** [comment]
Thanks @tobiasqueck @donal-tobin-sap
